### ISSUE TITLE
Realm generator rejects const values saying initialization must be const

### ISSUE THREAD

**Issue by derrickgw** [realm/realm-dart#1606]
### What happened?

Using constant classes or negative numbers used to work for initialization.  When upgrading from 1.6 to 2.0, this no longer works:

```[SEVERE] realm:realm_generator on lib/model/data/picture.dart:

Field initializers must be constant

in: package:sustain/model/data/picture.dart:19:14
    ╷
16  │ @RealmModel()
17  │ class _Example {
    │       ━━━━━━━━ in realm model for 'Example'
... │
19  │   double y = -1;
    │              ^^ Must be const
    ╵
Ensure the default value for field "y" is const

```

### Repro steps

Create a class with integer or float field and try to initialize it to a constant negative value.
Presumably this would work (or rather fail to work) with any other class that provides a const constructor.

### Version

2.0.0

### What Atlas Services are you using?

Local Database only

### What type of application is this?

Flutter Application

### Client OS and version

Linux

### Code snippets

```
@RealmModel()
class _Example {
  double x = 0;
  double y = -1;
  int z = -1;
}
```

### Relevant log output

```shell
[SEVERE] realm:realm_generator on lib/model/data/picture.dart:

Field initializers must be constant

in: package:sustain/model/data/picture.dart:19:14
    ╷
16  │ @RealmModel()
17  │ class _Example {
    │       ━━━━━━━━ in realm model for 'Example'
... │
19  │   double y = -1;
    │              ^^ Must be const
    ╵
Ensure the default value for field "y" is const


```

**Comment by Bilonik** [comment]
Hi,

Same issue here, any workaround?

**Comment by nirinchev** [comment]
We're aware of this and will address it with a future release. Currently, the only known mitigation is to downgrade to 1.6. I'm sorry for the regression.

**Comment by nielsenko** [comment]
Apparently `-1` is not a literal to the Dart analyzer

**Comment by derrickgw** [comment]
True, but even if it was, checking for literals is not a good solution.
Const variables are legal initializers, but are not literals.

**Comment by derrickgw** [comment]
That looks like a great improvement.  Unfortunately there are still some cases that cause false errors.  Here is an example:

```
import 'package:dart_numerics/dart_numerics.dart' as numerics;
const int myconst = 1;
@RealmModel(ObjectType.embeddedObject)
class $Test {
  ObjectId id = ObjectId();
  int value1 = myconst;
  int value2 = numerics.int64MinValue;
}
```

If I comment out the new const test, this generates legal dart.

**Comment by nielsenko** [comment]
@derrickgw Thank you for your observation. I have created #1612 to handle `value1` and `value2`.

But this const check was added explicitly to disallow `ObjectId id = ObjectId()`, so that is still not allowed. 

Consider:
```dart
@RealmModel()
class _Stuff {
  ObjectId id = ObjectId();
}
```

Which is legal with `realm_generator` 1.9.0.  If you look at the generated class, you will find that is illegal because it lifts the initializer expression and uses it as the default value of the optional parameter `id` on the generated constructor.
```dart
Stuff({
    ObjectId id = ObjectId(), // <-- this is not valid Dart, as default values must be const
  }) {
```
